src/js/components/tasks/path/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+
+import { usePrev } from '@/components/utils';
+import { Task } from '@/store/tasks/reducer';
+
+import Path from './path';
+import PathItem from './pathItem';
+
+interface TaskStatusPathProps {
+  task: Task;
+}
+const TaskStatusPath = ({ task }: TaskStatusPathProps) => {
+  const steps = [
+    { label: 'Planned', status: 'Planned' },
+    { label: 'In progress', status: 'In progress' },
+    { label: 'Review', status: 'In progress', pr: task.pr_is_open },
+    { label: 'Merged', status: 'Completed' },
+  ];
+  const [currentStep, setCurrentStep] = useState(0);
+  const [previousSteps, setPreviousSteps] = useState([]);
+  const stepsRef = usePrev(currentStep);
+  // add previous steps as complete
+  useEffect(() => {
+    if (currentStep >= 1) {
+      setPreviousSteps((oldArray) =>
+        [...oldArray, stepsRef].filter((item) => item !== currentStep),
+      );
+    }
```

Comment:
This works, I think, by taking the previous step, returned from `stepsRef` and adding it to this `previousSteps` array.So we can track when to move to the next step.